airflow/www_rbac/views.py

```diff
@@ -861,15 +854,35 @@ def blocked(self, session=None):
             })
         return wwwutils.json_response(payload)
 
-    @expose('/dagrun_success')
-    @has_access
-    @action_logging
-    def dagrun_success(self):
-        dag_id = request.args.get('dag_id')
-        execution_date = request.args.get('execution_date')
-        confirmed = request.args.get('confirmed') == 'true'
-        origin = request.args.get('origin')
+    def _mark_dagrun_state_as_failed(self, dag_id, execution_date, confirmed, origin):
```

Comment:
nit-picking again here:
`_mark_dagrun_state_as_success` and `_mark_dagrun_state_as_failed` share a lot of dupe code, maybe better to just use a common _mark_dagrun_state which can call `set_dag_run_state`

Reply:
This part was actually made in this way to follow the pattern of the methods you pointed out eariler: `set_dag_run_state_to_failed` and `set_dag_run_state_to_success`. I think it flows better to keep them in the same pattern but these two methods are more similar than `set_dag_run_state_to_failed` and `set_dag_run_state_to_success`--I'm fine to extract the common logic if you insist.